ppx_view/rewriter/rewriter.ml

```diff
@@ -238,6 +245,7 @@ and transl_pattern_desc loc desc =
     let expr_first,  vars_first  = transl_pattern first  in
     let expr_second, vars_second = transl_pattern second in
     let same_variables =
+      (* Isn't there an ordering issue? *)
```

Comment:
Okay I figured this one out, we need to have them in the same order otherwise we'd swap them in the `variables`.

I wish there was a way around that as it can be quite confusing to the unexperimented user. I'll try to make the error message clearer!